ESP8266/ESP8266.cpp

```diff
@@ -404,26 +406,36 @@ nsapi_error_t ESP8266::open_udp(int id, const char* addr, int port, int local_po
 
     _smutex.lock();
     if(local_port) {
-        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d,%d", id, type, addr, port, local_port)
-                && _parser.recv("OK\n");
+        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d,%d", id, type, addr, port, local_port);
     } else {
-        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d", id, type, addr, port)
-               && _parser.recv("OK\n");
+        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d", id, type, addr, port);
     }
 
     if (done) {
+        if (!_parser.recv("OK\n")) {
+            if (_sock_already) {
+                _sock_already = false; // To be raised again by OOB msg
+                done = close(id);
+                if (!done) { // Shouldn't happen
+                    goto EXIT;
+                }
+                _smutex.unlock();
+                return open_udp(id, addr, port, local_port); // Retry
```

Comment:
Unlimited recursion here.

Either do a loop, or add parameter to this local function which tell whether you are allowed to retry.

Reply:
For example `open_udp(int id, const char* addr, int port, int local_port, bool allow_retry);

Then first call with `true`, then on second call with `false`

Or use some other logic, like `goto RETRY` with internal variable.
Unlimited retrying not allowed either.

Reply:
Also, on the second loop, this should cause `MBED_ERROR` because we did not fix the issue by closing the socket.